module/os/freebsd/spl/spl_taskq.c

```diff
@@ -192,13 +192,12 @@ taskq_remove(taskq_ent_t *ent)
 {
 	taskqid_t tqid = ent->tqent_id;
 
-	if (!ent->tqent_registered)
+	if (tqid == 0)
 		return;
-
 	sx_xlock(TQIDHASHLOCK(tqid));
 	CK_LIST_REMOVE(ent, tqent_hash);
 	sx_xunlock(TQIDHASHLOCK(tqid));
-	ent->tqent_registered = B_FALSE;
+	ent->tqent_id = 0;
```

Comment:
This does not look legit. 2 parallel calls here can result in an attempt to unlink the element *twice*.

Unlinking before whacking the last ref is pretty weird on its own, but maybe this is bit is fine.

So bare minimum this needs to check again with the lock held.

Reply:
You seem to be right, though it seems the problem was there before me too.  Will fix.